src/rpc/dial.rs

```diff
@@ -365,39 +379,132 @@ async fn get_auth_token(
 }
 
 impl DialBuilder<WithCredentials> {
-    /// attempts to establish a connection with credentials to the DialBuilder's given uri
-    pub async fn connect(self) -> Result<AddAuthorization<ViamChannel>> {
+    fn duplicate_uri(&self) -> Option<Parts> {
+        match &self.config.uri {
+            None => None,
+            Some(parts) => {
+                let uri = Uri::builder()
+                    .authority(parts.authority.clone()?)
+                    .path_and_query(parts.path_and_query.clone()?)
+                    .scheme(parts.scheme.clone()?);
+                Some(uri.build().ok()?.into_parts())
+            }
+        }
+    }
+
+    async fn get_mdns_uri(&self) -> Option<Parts> {
+        if self.config.disable_mdns {
+            return None;
+        }
+
+        log::debug!("Attempting to find address via mDNS");
+
+        let mut uri = self.duplicate_uri()?;
+        let candidate = uri.authority.clone()?.to_string();
+        let candidates = vec![candidate.replace(".", "-"), candidate.to_string()];
+
+        let stream = match mdns::discover::all(SERVICE_NAME, Duration::from_secs(1)) {
```

Comment:
would using mdns::resolve::one make more sense here? we could just target a specific robot that way

Reply:
I tried it briefly, but found that the query was successful when using `discover::all` but unsuccessful when using `resolve::one`. I don't entirely know why this is, but suspect it's because `resolve::one` only allows a single query, but the first query doesn't seem to guarantee that we will find what we're looking for? I could be wrong, but enforcing only a single query seems to be the only difference between `resolve::one` and `discover::all` ( `resolve::one` is just doing a `discover::all` and then matching on `hostname` under the hood).

Reply:
i'm not surprised :) we should leave it this way then!